go/apps/ctrl/services/deployment/deploy_workflow.go

```diff
@@ -88,8 +93,34 @@ func (w *DeployWorkflow) Run(ctx hydra.WorkflowContext, req *DeployRequest) erro
 		"workspace_id", req.WorkspaceID,
 		"project_id", req.ProjectID,
 	)
+
+	workspace, err := hydra.Step(ctx, "get-workspace", func(stepCtx context.Context) (db.Workspace, error) {
+		return db.Query.FindWorkspaceByID(stepCtx, w.db.RW(), req.WorkspaceID)
+	})
+	if err != nil {
+		return err
+	}
+	project, err := hydra.Step(ctx, "get-project", func(stepCtx context.Context) (db.FindProjectByIdRow, error) {
+		return db.Query.FindProjectById(stepCtx, w.db.RW(), req.ProjectID)
+	})
+	if err != nil {
+		return err
+	}
+	environment, err := hydra.Step(ctx, "get-environment", func(stepCtx context.Context) (db.FindEnvironmentByIdRow, error) {
+		return db.Query.FindEnvironmentById(stepCtx, w.db.RW(), req.EnvironmentID)
+	})
+	if err != nil {
+		return err
+	}
+	deployment, err := hydra.Step(ctx, "get-deployment", func(stepCtx context.Context) (db.FindDeploymentByIdRow, error) {
+		return db.Query.FindDeploymentById(stepCtx, w.db.RW(), req.DeploymentID)
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
can we please join this in a single query

Reply:
hmm what's the benefit?
this is async anyways and 200ms more don't hurt anyone

I felt like this is nicer than some special one-off query, but I don't have a strong opinion here

Reply:
ok, will do